flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/IntermediateResultPartitionTest.java

```diff
@@ -140,6 +150,126 @@ public void testBlockingPartitionResetting() throws Exception {
+    }
+
+    private void testGetNumberOfSubpartitions(
+            int consumerParallelism,
+            DistributionPattern distributionPattern,
+            boolean isDynamicGraph,
+            Collection<Integer> expectedNumSubpartitions)
+            throws Exception {
+
+        final int producerParallelism = 2;
+        final int consumerMaxParallelism = 13;
+
+        final ExecutionGraph eg =
+                createExecutionGraph(
+                        producerParallelism,
+                        consumerParallelism,
+                        consumerMaxParallelism,
+                        distributionPattern,
+                        isDynamicGraph);
+
+        final Iterator<ExecutionJobVertex> vertexIterator =
+                eg.getVerticesTopologically().iterator();
+        final ExecutionJobVertex producer = vertexIterator.next();
+
+        if (isDynamicGraph) {
+            ExecutionJobVertexTest.initializeVertex(producer);
+        }
+
+        final IntermediateResult result = producer.getProducedDataSets()[0];
+
+        assertThat(expectedNumSubpartitions.size(), is(producerParallelism));
+        assertThat(
+                Arrays.stream(result.getPartitions())
+                        .map(IntermediateResultPartition::getNumberOfSubpartitions)
+                        .collect(Collectors.toList()),
+                equalTo(expectedNumSubpartitions));
+    }
+
+    private static ExecutionGraph createExecutionGraph(
+            int producerParallelism,
+            int consumerParallelism,
+            int consumerMaxParallelism,
+            DistributionPattern distributionPattern,
+            boolean isDynamicGraph)
+            throws Exception {
+
+        final JobVertex v1 = new JobVertex("v1");
+        v1.setInvokableClass(NoOpInvokable.class);
+        v1.setParallelism(producerParallelism);
+
+        final JobVertex v2 = new JobVertex("v2");
+        v2.setInvokableClass(NoOpInvokable.class);
+        if (consumerParallelism > 0) {
+            v2.setParallelism(consumerParallelism);
+        }
+        if (consumerMaxParallelism > 0) {
+            v2.setMaxParallelism(consumerMaxParallelism);
+        }
+
+        v2.connectNewDataSetAsInput(v1, distributionPattern, ResultPartitionType.BLOCKING);
+
+        final JobGraph jobGraph =
+                JobGraphBuilder.newBatchJobGraphBuilder()
+                        .addJobVertices(Arrays.asList(v1, v2))
+                        .build();
+
+        final Configuration configuration = new Configuration();
+
+        return TestingDefaultExecutionGraphBuilder.newBuilder()
+                .setJobGraph(jobGraph)
+                .setJobMasterConfig(configuration)
+                .setVertexParallelismStore(
+                        computeVertexParallelismStoreConsideringDynamicGraph(
+                                jobGraph.getVertices(), isDynamicGraph, consumerMaxParallelism))
+                .buildDynamicGraph();
```

Comment:
I guess we should not `buildDynamicGraph` if `isDynamicGraph` is false?

Reply:
And one question comes to me that why the above tests can pass?

Reply:
Thanks for pointing that out, and the `testGetNumberOfSubpartitionsForNonDynamicAllToAllGraph` and `testGetNumberOfSubpartitionsForNonDynamicPointwiseGraph` don't pass in 73c2ecd546c2ad9be7ee20382716edc9f6c6c574. I will fix it.